packages/dds/tree/src/simple-tree/treeApi.ts

```diff
@@ -142,11 +177,74 @@ export const treeNodeApi: TreeNodeApi = {
 
 /**
  * A collection of events that can be raised by a {@link TreeNode}.
+ *
+ * @privateRemarks
+ * TODO: add a way to subscribe to a specific field (for afterShallowChange and afterDeepChange).
+ * Probably have object node and map node specific APIs for this.
+ *
+ * TODO: ensure that subscription API for fields aligns with API for subscribing to the root.
+ *
+ * TODO: add more wider area (avoid needing tons of afterShallowChange registration) events for use-cases other than afterDeepChange.
+ * Some ideas:
+ *
+ * - afterDeepChange, but with some subtrees/fields/paths excluded
+ * - helper to batch several afterShallowChange calls to a afterDeepChange scope
+ * - parent change (ex: registration on the parent field for a specific index: maybe allow it for a range. Ex: node event takes optional field and optional index range?)
+ * - new content inserted into subtree. Either provide event for this and/or enough info to afterDeepChange to find and search the new sub-trees.
+ * Add separate (non event related) API to efficiently scan tree for given set of types (using low level cursor and schema based filtering)
+ * to allow efficiently searching for new content (and initial content) of a given type.
+ *
  * @public
  */
-export interface TreeNodeEvents {
+export interface TreeChangeEvents {
 	/**
-	 * Raised on a node right after a change is applied to one of its fields or the fields of a descendant node.
+	 * Emitted by a node when:
+	 * - The content of one or more of the node's properties changes (i.e., the property now contains a new node, or
+	 * nothing if it previously contained a node).
+	 * - For an array node, when an element is added, removed, or moved.
+	 * - For a map node, when a key is added, updated, or removed.
```

Comment:
To make sure this renders correctly as a list in the generated docs, I recommend:
```suggestion
	 * Emitted by a node when:
	 
	 * - The content of one or more of the node's properties changes (i.e., the property now contains a new node, or
	 * nothing if it previously contained a node).
	 
	 * - For an array node, when an element is added, removed, or moved.
	 
	 * - For a map node, when a key is added, updated, or removed.
```

I'm hoping to add some smarts to api-markdown-documenter to do better here, but technically lists like this are not part of the TSDoc spec. https://github.com/microsoft/FluidFramework/wiki/TSDoc-Guidelines#line-breaks